src/box/lua/net_box.lua

```diff
@@ -1201,6 +1242,10 @@ index_metatable = function(remote)
         if opts and opts.buffer then
             error("index:count() doesn't support `buffer` argument")
         end
+        if self.space.name == nil or self.name == nil then
+            error("index:count() is available only when space and index " ..
+                  "names are set")
+        end
```

Comment:
@alyapunov insists that this should be fixed now, in a separate patch. Please also open an issue (unless we don't have one already) because this is a user-visible improvement (`call` will start accepting numeric indexes).

Reply:
Opened #8604 and added a patch that closes it. Patched `net.box` to use this feature.